artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireConnection.java

```diff
@@ -832,7 +863,16 @@ public AMQConnectionContext initContext(ConnectionInfo info) throws Exception {
       createInternalSession(info);
 
       // the actor can only be used after the WireFormat has been initialized with versioning
-      this.openWireActor = new ThresholdActor<>(executor, this::act, actorThresholdBytes, OpenWireConnection::getSize, this::disableAutoRead, this::flushedActor);
+      this.openWireActor = new ThresholdActor<>(executor, this::act, actorThresholdBytes, OpenWireConnection::getSize, this::disableAutoRead, this::flushedActor) {
+         @Override
+         protected void doTask(Object task) {
```

Comment:
I would place this logic on ThresholdActor.

Set something when shutdown is called, and have the doTask checking if (task==shutdownTask), it would do it right away.

The ThresholdActor is being done for OpenWire pretty much, and I think that logic would apply to the super class anyways.

Reply:
basically.. just keep doTask a final, and bring that logic up there. I think it would be clearer that way!?

Reply:
fair enough, as openwire is the only user of that class. on it. thanks for the review